packages/react-kit/src/hooks/contracts/erc721/useErc721TokenUri.ts

```diff
@@ -0,0 +1,28 @@
+import { useQuery } from "react-query";
+import { useCoreSDKWithContext } from "../../core-sdk/useCoreSdkWithContext";
+
+export const useErc721TokenUri = (
+  props: {
+    contractAddress: string | undefined;
+    tokenIds: (string | null | undefined)[] | undefined;
+  },
+  { enabled }: { enabled: boolean | undefined }
+) => {
+  const coreSDK = useCoreSDKWithContext();
+  return useQuery(["useErc721TokenUri", coreSDK.uuid, props], async () => {
+    const { contractAddress, tokenIds } = props;
+    if (!contractAddress || !tokenIds) {
+      return;
+    }
+    return Promise.all(
+      tokenIds.map((tokenId) =>
+        tokenId
+          ? coreSDK.erc721TokenUri({
+              contractAddress,
+              tokenId
+            })
+          : null
+      )
+    );
+  });
```

Comment:
```suggestion
  return useQuery(
    ["useErc721TokenUri", coreSDK.uuid, props],
    async () => {
      const { contractAddress, tokenIds } = props;
      if (!contractAddress || !tokenIds) {
        return;
      }
      return Promise.all(
        tokenIds.map((tokenId) =>
          tokenId
            ? coreSDK.erc721TokenUri({
                contractAddress,
                tokenId
              })
            : null
        )
      );
    },
    { enabled }
  );
```

Reply:
And the same change is also required in
- packages/react-kit/src/hooks/contracts/erc721/useErc721OwnerOf.ts
- packages/react-kit/src/hooks/offer/useExchangeTokenBalance.ts